### ISSUE TITLE
StackOverflowError on an `utils` function

### ISSUE THREAD

**Issue by zhanglw0521** [ACEsuit/EquivariantModels.jl#28]
In `utils.jl`, we have the following 

```julia
function specnlm2spec1p(spec_nlm)
    spec1p = union(spec_nlm...)
    lmax = [ spec1p[i].l for i = 1:length(spec1p) ] |> maximum
    nmax = [ spec1p[i].n for i = 1:length(spec1p) ] |> maximum
    return spec1p, lmax, nmax + 1
end
```
which extracts the corresponding `Aspec` from an input `AAspec` (I actually don't remember why we need lmax and nmax in the output but let's keep it for now).

However the `union` function in the first line seems to give a StackOverflowError. For instance, when we run 
```julia
Zi = :C
Zs = [:H,:C,:O]

cats_ext = [(Zi,Z) for Z in Zs] |> unique

maxdeg = 6
ord = 3
radial = simple_radial_basis(legendre_basis(maxdeg))
Lmax = 0

Aspec, AAspec = degord2spec(radial; totaldegree = maxdeg, 
                                  order = ord, 
                                  Lmax = Lmax, catagories = cats_ext)
```
this issue arises. 

It can be fixed by something like
```julia
function specnlm2spec1p(spec_nlm)
    spec1p = []
    for (i, spec_nlm_i) in enumerate(spec_nlm)
        push!(spec1p, spec_nlm_i...)
        unique!(spec1p)
    end
    lmax = [ spec1p[i].l for i = 1:length(spec1p) ] |> maximum
    nmax = [ spec1p[i].n for i = 1:length(spec1p) ] |> maximum
    return spec1p, lmax, nmax + 1
end
```

A PR regarding this issue will follow soon.

**Comment by cortner** [comment]
Do you understand why we implemented it with a `union` in the past? If not, then in the PR, please leave the original implementation,  comment it out and add a remark that we don't understand why the original implementation was done this way.  Then we can more easily recover it if a different use-case fails in the near future.

**Comment by cortner** [comment]
one possibility is that it's just a matter of size of the input array - would you mind testing this?

**Comment by zhanglw0521** [comment]
resolved by #29
